frame m added to pipeline

aaarchmiller opened the pull request:
This is a relatively simple add in for now to make "frame_m" (m.png) for the website build. Frame m is the overview of the last 70 years by method. 

Running `tar_make()` should 
1. rebuild frames k and l with transparent axes, since streamflow is no longer the y-axis (y-axis is essentially dimensionless for the last three frames) *and* 
2. it will create the new m.png. 
3. In addition, I added the tag and text to `src/assets/text/scrollyText.js` for frame m so that it will run on the website build. 

After `tar_make()` the website can be built (with node v 14.17.0) and it should have frame m now appear at the end of the viz. 

_Awareness:_
There is some pre-processing of the data that gets read into the `frame_m()` function to make this frame. My next PR is going to be to work on better modularizing the entire pipeline, and during that phase, I will remove any redundancies in coding *and* make sure to remove any pre-processing steps from any of the frames to the `p2_targets`.

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
-
+frame_m <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+                    p2_1951_2020_drought_prop_site, p2_1951_2020_drought_prop_jd_7d,
+                    bottom_bars, canvas, inset, out_png){
```
RC: The function definition here has 8 arguments, but when it's called in `3_visualize.R` it only has 7 (no `bottom_bars`)

hcorson-dosch-usgs commented on src/assets/text/scrollyText.js:
```diff
@@ -51,6 +51,10 @@ export default {
+        },
+        {
+          id: 'm',
+          text: 'text l: And even more recognizable across the past 70 years.'
```
```suggestion
          text: 'text m: And even more recognizable across the past 70 years.'
```

hcorson-dosch-usgs replied:
^ RC

aaarchmiller commented:
I just tested the website with `npm install` first, and there was a new package lock. I just committed it and pushed it up to go into this PR

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+  
+  # Select correct station ID and threshold
+  p2_droughts_site <- p2_1951_2020_drought_prop_site %>%
+    filter(StaID == '03221000', threshold == 10) %>%
```
SC: When you do your refactoring, I'd either a) provide the station id and threshold value as arguments (see my note below re:  modularizing and separating the data prep steps) or b) define them as targets in `2_process.R` , in which case they could be referenced to build `p2_droughts_learner_viz_df` and the other dataframes for the learner viz, and when prepping the data for this plot.

Since this pipeline is so customized to one site, and that is unlikely to change, I don't think b) is necessary, but I do think it would be good for a user to be able to see in the target recipe files which site and which threshold is being used for the plots, whether that's as function arguments (a) or standalone targets (b). As is, that information (for this plot) is buried in this function.

aaarcher-usgs replied:
Added as a comment to issue #14

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+                    bottom_bars, canvas, inset, out_png){
+  
+  # Select correct station ID and threshold
+  p2_droughts_site <- p2_1951_2020_drought_prop_site %>%
```
RC:  I'd avoid using this `targets` naming convention (prefixed with `p#_`) for variables that are defined within functions, as it may lead to confusion about which objects are and are not targets.

aaarcher-usgs replied:
Makes sense!

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+    filter(start_year == end_year)
+  p2_droughts_site <- bind_rows(p2_droughts_site_wrapYr,
+                                p2_droughts_site_noWrapYr)
+  
```
SC: This may well be on your to-do list, but when you refactor, I think all of the code above this line, which is repeated for each dataset, should be pulled into a separate data-prep function that can be called once for each dataset, with the raw dataset, method_name, site, threshold value, and placeholder year (1963) as arguments. Since it is data prep, I think it could potentially also be moved to `2_process.R`. That would a) make the data prep more modular and reduce repeated code, and b) keep this function about generating the plot

aaarcher-usgs replied:
Added to issue #14

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+          axis.ticks.y = element_line(color = "transparent"))
+  
+  
+  # Bottom bars
```
RC: Maybe rename these now that they are no longer bottom bars

aaarcher-usgs replied:
These bottom bars are still underneath this image as bottom bars, although not sure what value they add to these later viz frames

hcorson-dosch-usgs replied:
oooh sorry I mistook this section. Looks like they aren't actually added to the plot, so this section could maybe be deleted? At least I don't see them specified in lines 1017-1039 and don't see them in the plot on the website:
![image](https://user-images.githubusercontent.com/54007288/185680433-5e9c9e4b-e87f-4348-9108-0ac04c35f307.png)

aaarcher-usgs replied:
They were supposed to still be on there for now, so I fixed it so they show correctly. Thanks for noticing!

hcorson-dosch-usgs commented on 3_visualize/src/drought_learner_viz_components.R:
```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+              y = 0.1,
+              height = 0.9,
+              width = 1)+
+    # Inset map
```
RC: I think the inset map should either be move or dropped here, since it's blocking some of the data

hcorson-dosch reviewed (changes requested):
Thanks for adding this last plot, Althea. It's a nice addition. I just have a few requested changes (noted with 'RC:' at the start of the comment. Once those are made I think this PR is good to go. See also my suggested changes (prefixed with 'SC:'), which I think could be addressed in your next refactoring PR.

aaarchmiller commented:
@hcorson-dosch Thanks for the great PR review! I addressed all of your feedback and updated the repo to match. 

1. I fixed variable naming issue so that the function variables aren't the same as the target names and don't contain `pX_`. Note that I changed this in the `p3_` targets AND the `p2_` targets. 
2. I added bottom bars back into frame m
3. I DIDN'T completely modularize the pipeline yet, but I did add several of your comments to Issue #14 so that I can address those in the next branch.

Thanks!

hcorson-dosch reviewed (approved):
Looks great! Thanks for making those changes.
